### PR TITLE
Feat x86 docker

### PR DESCRIPTION
**Pull request by mdegans**
* slightly smaller apt layer
* use gpu support where possible (with `--gpus all`)

### CODE REVIEW

**Comment by mhejrati** [comment]
Shall we create one with GPU and one without GPU support?  The GPU base docker is almost 1GB larger than the one without GPU.

File: applications/smart-distancing/Dockerfile-x86 [hunk 1]
```diff
@@ -1,9 +1,14 @@
...
 VOLUME  /repo
 WORKDIR /repo/applications/smart-distancing
 
-RUN apt-get update && apt-get install -y pkg-config libsm6 libxext6 libxrender-dev
+RUN apt-get update && apt-get install -y --no-install-recommends \
+        libsm6 \
+        libxext6 \
+        libxrender-dev \
+        pkg-config \
+    && rm -rf /var/lib/apt/lists/*
```
Comment (mohammad7t) [inline applications/smart-distancing/Dockerfile-x86:11]: adding `&& apt-get clean` further reduces the image size

**Comment by mdegans** [comment]
> Shall we create one with GPU and one without GPU support? The GPU base docker is almost 1GB larger than the one without GPU.

Sounds good. The Cuda stuff adds a lot.

File: applications/smart-distancing/Dockerfile-x86 (hunk 1 above)
Comment (mdegans) [reply applications/smart-distancing/Dockerfile-x86:11]: That used to be true, but newer Debian and Ubuntu images automatically run it at the end of layer.

see note at bottom of run section:
https://docs.docker.com/develop/develop-images/dockerfile_best-practices/#run

I have no idea why they didn't go the extra mile and nuke the apt lists as well. It seems like an oversight.

Comment (mohammad7t) [reply applications/smart-distancing/Dockerfile-x86:11]: Aha! thanks.

**Comment by mhejrati** [comment]
@mdegans happy to change this PR according to the convo above or wait for you to do it, let me know if you are working on it.

**Comment by mdegans** [comment]
Sounds good. I will focus on Nginx tomorrow then and the hlssink, I think? Somebody else suggested that in another thread.

Re: other thread's mistaken post. I meant to reply here. It might be worth running the docker build with and without (apt clean) to double check. Sometimes documentation is wrong and it does say 'official images'.

**Comment by mhejrati** [comment]
@mdegans @mhsekhavat is this PR still relevant?

**Comment by mdegans** [comment]
@mhejrati Probably not since we're using the new repo now. I'll probably re-submit these changes there with the suggested no-gpu and gpu versions in separate dockerfiles.

**Pull request closed without merging**